<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Summer Institute of Statical Genetics (Module QG3)" />

<meta name="date" content="2025-06-10" />

<title>Interaction testing and vQTL mapping</title>

<script src="site_libs/header-attrs-2.22/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SISG 2025 QG3</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Sessions
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Day 1</li>
    <li>
      <a href="QG3_Association_Testing_practical.html">Session 1</a>
    </li>
    <li>
      <a href="QG3_Plink_Population_Structure_practical.html">Session 2</a>
    </li>
    <li>
      <a href="QG3_Relatedness_REGENIE_practical.html">Session 3</a>
    </li>
    <li>
      <a href="QG3_RV_tests_practical.html">Session 4</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Day 2</li>
    <li>
      <a href="QG3_Beyond_Standard_GWAS_practical.html">Session 5</a>
    </li>
    <li>
      <a href="QG3_Downstream-Analyses_practical.html">Session 6</a>
    </li>
    <li>
      <a href="QG3_Polygenic_Scores_practical.html">Session 7</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Day 3</li>
    <li>
      <a href="QG3_Power-Design_practical.html">Session 9</a>
    </li>
    <li>
      <a href="QG3_CC_Imbalance_practical.html">Session 10</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/joellembatchou/SISG2025_Association_Mapping">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Interaction testing and vQTL mapping</h1>
<h4 class="author">Summer Institute of Statical Genetics (Module
QG3)</h4>
<h4 class="date">2025-06-10</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-06-10
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>SISG2025_Association_Mapping/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is ignored by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230530code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230530)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230530code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230530)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjoellembatchouSISG2025AssociationMappingtreee43c81505dc79c2af0c06d6ef99aba63f3a59f92targetblanke43c815a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/joellembatchou/SISG2025_Association_Mapping/tree/e43c81505dc79c2af0c06d6ef99aba63f3a59f92" target="_blank">e43c815</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomjoellembatchouSISG2025AssociationMappingtreee43c81505dc79c2af0c06d6ef99aba63f3a59f92targetblanke43c815a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/joellembatchou/SISG2025_Association_Mapping/tree/e43c81505dc79c2af0c06d6ef99aba63f3a59f92" target="_blank">e43c815</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .qodo/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/QG3_Association_Testing_practical_Key.Rmd
    Ignored:    analysis/QG3_Beyond_Standard_GWAS_practical_Key.Rmd
    Ignored:    analysis/QG3_CC_Imbalance_practical_Key.Rmd
    Ignored:    analysis/QG3_Downstream-Analyses_practical_Key.Rmd
    Ignored:    analysis/QG3_Plink_Population_Structure_practical_Key.Rmd
    Ignored:    analysis/QG3_Polygenic_Scores_practical_Key.Rmd
    Ignored:    analysis/QG3_Power-Design_practical_Key.Rmd
    Ignored:    analysis/QG3_RV_tests_practical_Key.Rmd
    Ignored:    analysis/QG3_Relatedness_REGENIE_practical_Key.Rmd
    Ignored:    data/run_regenie.r
    Ignored:    data/sim_rels_geno.bed
    Ignored:    exe/
    Ignored:    lectures/
    Ignored:    mk_website.R
    Ignored:    notes.txt
    Ignored:    tmp/

Untracked files:
    Untracked:  .mk_website.R.swp
    Untracked:  _workflowr.yml
    Untracked:  analysis/.Rhistory
    Untracked:  analysis/.index.Rmd.swp
    Untracked:  analysis/QG3_Plink_Population_Structure_practical_Key_cache/

Unstaged changes:
    Modified:   analysis/QG3_Beyond_Standard_GWAS_practical.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/QG3_Beyond_Standard_GWAS_practical_Key.Rmd</code>) and
HTML (<code>docs/QG3_Beyond_Standard_GWAS_practical_Key.html</code>)
files. If you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/joellembatchou/SISG2025_Association_Mapping/5949e8a093c7e926391a8d3f70c194f137471340/docs/QG3_Beyond_Standard_GWAS_practical_Key.html" target="_blank">5949e8a</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2025-06-10
</td>
<td>
Build site.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Before you begin:</p>
<p>Make sure that R is installed on your computer For this lab, we will
use the following R libraries:</p>
<pre class="r"><code>library(qqman)</code></pre>
<p>This practical aims at illustrating the relationship between tests
for vQTLs and tests for interaction effects. We provide a set of
<code>R</code> commands below to simulate the phenotypes, genotypes and
covariates (at <span class="math inline">\(M=1000\)</span> SNPs) of
<span class="math inline">\(N=10,000\)</span> samples and 2 covariates.
We will chose the first of the 10 SNPs to be causal (either QTL, vQTL or
both). We generate two phenotypes which differ based on whether the
causal variant influences the phenotypic variance of the trait directly
or influences the phenotypic mean through GxE effects.</p>
<p><span class="math display">\[
E(Y_{vQTL}) = G\beta,\; Var(Y_{vQTL}) = 1 + G\alpha
\]</span></p>
<p><span class="math display">\[
E(Y_{gxe}) = G\beta + E\gamma +  (G\times E)\alpha
\]</span></p>
<div id="simulated-dataset" class="section level2">
<h2>Simulated dataset</h2>
<p><strong>Copy/Run the following command to enable the
<code>simulate_vqtl_data</code> function in your current <code>R</code>
environment.</strong></p>
<pre class="r"><code>set.seed(646909) # For reproducibility
n_individuals &lt;- 10000
n_snps &lt;- 1000

# Function to simulate genotype, phenotype, and covariate data
# Mean effect != 0 corresponds to QTL
# Variance effect != 0 corresponds to vQTL
simulate_vqtl_data &lt;- function(n_individuals = 10000, n_snps = 1000, mean_effect = 1, variance_effect = 2) {

  # Simulate genotype data with no LD (0, 1, 2 for SNPs)
  genotype_data &lt;- as.data.frame(matrix(sample(0:2, n_individuals * n_snps, replace = TRUE), 
                                        nrow = n_individuals, ncol = n_snps))
  colnames(genotype_data) &lt;- paste0(&quot;SNP&quot;, 1:n_snps)

  # Simulate environmental covariate (e.g., smoking status)
  covariate_data &lt;- data.frame(
    environment = factor(sample(c(&quot;non-smoker&quot;, &quot;smoker&quot;), n_individuals, replace = TRUE), levels = c(&quot;non-smoker&quot;, &quot;smoker&quot;)),
    age = runif(n_individuals, 40,60)
  )

  # Approach 1: Modeling SNP effect in the phenotypic variance
  phenotype_variance &lt;- rnorm(n_individuals, 
                              mean = mean_effect * genotype_data$SNP1, 
                              sd = 1 + variance_effect * genotype_data$SNP1)

  # Approach 2: Modeling GxE effect in the phenotypic mean
  phenotype_gxe &lt;- rnorm(n_individuals, 
                         mean = mean_effect * genotype_data$SNP1 + 
                                variance_effect * as.numeric(covariate_data$environment) *genotype_data$SNP1, 
                         sd = 1)

  # Combine phenotypes into a data frame
  phenotype_data &lt;- data.frame(Y_vQTL = phenotype_variance, Y_gxe = phenotype_gxe)

  # Return a list of data frames
  return(list(genotype = genotype_data, phenotype = phenotype_data, covariate = covariate_data))

}</code></pre>
<p>The <code>simulate_vqtl_data</code> function has 4 input parameters:
<code>n_individuals</code> (sample size), <code>n_snps</code> (number of
SNPs tested), <code>mean_effect</code> (the effect size of the causal
variant on the phenotypic mean), <code>variance_effect</code> (the
effect size of the causal variant on the phenotypic variance
<code>for phenotype Y1</code> OR the size of the GxE interaction effect
<code>for phenotype Y2</code>).</p>
<p>Let’s run a first data set where the causal variant has no effect on
the trait, i.e., <code>mean_effect = 0</code> and
<code>variance_effect = 0</code>.</p>
<pre class="r"><code>simulated_data &lt;- simulate_vqtl_data(mean_effect = 0, variance_effect = 0)</code></pre>
<p>Note: you can use <code>str(simulated_data)</code> to see a snipper
of what’s returned by the <code>simulate_vqtl_data</code> function.</p>
<p>We can visualize the phenotype distribution across genotype groups
for SNP1.</p>
<pre class="r"><code>trait_name &lt;- &quot;Y_vQTL&quot;
boxplot(phenotype[,trait_name] ~ genotype[,&quot;SNP1&quot;], data = simulated_data,
        xlab = &quot;Genotype&quot;,
        ylab = trait_name,
        col = c(&quot;#1f77b4&quot;, &quot;#ff7f0e&quot;, &quot;#2ca02c&quot;))</code></pre>
<p><img src="figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/joellembatchou/SISG2025_Association_Mapping/blob/5949e8a093c7e926391a8d3f70c194f137471340/docs/figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-3-1.png" target="_blank">5949e8a</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2025-06-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/joellembatchou/SISG2025_Association_Mapping/blob/648aef41f15951e1f07b9b9389ba1f963cc6467f/docs/figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-3-1.png" target="_blank">648aef4</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2025-06-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="testing-for-mean-qtl" class="section level2">
<h2>Testing for (mean) QTL</h2>
<p>We use of <code>lm()</code> in R to assess phenotypic mean
differences across genotypes for each variant.</p>
<p><strong>Copy/Run the following command to enable the
<code>lm_test</code> function in your current <code>R</code>
environment.</strong></p>
<pre class="r"><code># This function applies linear regression test to each of the simulated variant for a given trait
# trait_name = {&quot;Y_vQTL&quot;, &quot;Y_gxe&quot;}
lm_test &lt;- function(data, trait_name) {
  
  p_values &lt;- rep(NA, n_snps)
  names(p_values) &lt;- colnames(data$genotype)

  for(i in 1:n_snps){
    p_values[i] &lt;- summary(lm(phenotype[, trait_name] ~ genotype[,i], data = data))$coef[2,4]
  }
  
  # Return the p-values
  return(p_values)
}</code></pre>
<p>Based on the parameters used in the simulation, would we expect
significant result when testing the causal SNP1?</p>
<p>First we do it for trait “Y_vQTL” (do it again on your own for the
other trait “Y_gxe”)</p>
<pre class="r"><code>trait_name &lt;- &quot;Y_vQTL&quot;
qtl_pvals &lt;- lm_test(data = simulated_data, trait_name =  trait_name)

qtl_pvals[1] # P-value of the causal SNP1</code></pre>
<pre><code>     SNP1 
0.2942319 </code></pre>
<p>You can make a qq plot to visualize the p-values highlighting the
causal SNP1.</p>
<p><strong>Copy/Run the following command to enable the
<code>plot_qq</code> function in your current <code>R</code>
environment.</strong></p>
<pre class="r"><code>plot_qq &lt;- function(pvals){
  qq(pvals)
  points(-log10(ppoints(n_snps)[rank(pvals)[1]]), -log10(pvals[1]),
       col = &quot;blue&quot;, pch = 19, cex = 1.5)
}</code></pre>
<pre class="r"><code>plot_qq(qtl_pvals)</code></pre>
<p><img src="figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/joellembatchou/SISG2025_Association_Mapping/blob/5949e8a093c7e926391a8d3f70c194f137471340/docs/figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-7-1.png" target="_blank">5949e8a</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2025-06-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="testing-for-vqtl" class="section level2">
<h2>Testing for vQTL</h2>
<p>We make use of Levene’s test to assess phenotypic variance
differences across genotypes for each variant.</p>
<p><strong>Copy/Run the following command to enable the
<code>levene_test</code> function in your current <code>R</code>
environment.</strong></p>
<pre class="r"><code># This function applies levene test to each of the simulated variant for a given trait
# trait_name = {&quot;Y_vQTL&quot;, &quot;Y_gxe&quot;}
levene_test &lt;- function(data, trait_name) {
  
  p_values &lt;- rep(NA, n_snps)
  names(p_values) &lt;- colnames(data$genotype)
  
  for(i in 1:n_snps){
    # Ensure the genotype is a factor
    factor_geno &lt;- as.factor(data$genotype[,i])
  
    # Calculate the absolute deviations from the group means
    abs_dev &lt;- abs(data$phenotype[, trait_name] - ave(data$phenotype[, trait_name], factor_geno, FUN = mean))

    # Perform an ANOVA on the absolute deviations
    p_values[i] &lt;- summary(aov(abs_dev ~ factor_geno))[[1]]$`Pr(&gt;F)`[1]
  }
  
  # Return the p-values
  return(p_values)
}</code></pre>
<p>Based on the parameters used in the simulation, would we expect
significant result when testing the causal SNP1?</p>
<pre class="r"><code>trait_name &lt;- &quot;Y_vQTL&quot;
levene_pvals &lt;- levene_test(data = simulated_data, trait_name =  trait_name)  # Levene&#39;s test for variance differences

levene_pvals[1] # P-value of the causal SNP1</code></pre>
<pre><code>     SNP1 
0.2343523 </code></pre>
<p>You can make a qq plot to visualize the p-values highlighting the
causal SNP1.</p>
<pre class="r"><code>plot_qq(levene_pvals)</code></pre>
<p><img src="figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/joellembatchou/SISG2025_Association_Mapping/blob/5949e8a093c7e926391a8d3f70c194f137471340/docs/figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-10-1.png" target="_blank">5949e8a</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2025-06-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="testing-for-gxe" class="section level2">
<h2>Testing for GxE</h2>
<p>We again use of <code>lm()</code> in R but this time also include an
interaction term with the environment.</p>
<p><strong>Copy/Run the following command to enable the
<code>gxe_test</code> function in your current <code>R</code>
environment.</strong></p>
<pre class="r"><code>gxe_test &lt;- function(data, trait_name){
  pvals &lt;- rep(NA, n_snps)
  names(pvals) &lt;- colnames(data$genotype)

  for(i in 1:n_snps){
    pvals[i] &lt;- summary(lm(phenotype[, trait_name] ~ genotype[,i]*covariate$environment, data = data))$coef[4,4]
  }
  
  return(pvals)
}</code></pre>
<pre class="r"><code>trait_name &lt;- &quot;Y_vQTL&quot;
gxe_pvals &lt;- gxe_test(data = simulated_data, trait_name =  trait_name)
gxe_pvals[1] # P-value of the causal SNP1</code></pre>
<pre><code>     SNP1 
0.4361276 </code></pre>
<p>You can make a qq plot to visualize the p-values highlighting the
causal SNP1.</p>
<pre class="r"><code>plot_qq(gxe_pvals)</code></pre>
<p><img src="figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/joellembatchou/SISG2025_Association_Mapping/blob/5949e8a093c7e926391a8d3f70c194f137471340/docs/figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-13-1.png" target="_blank">5949e8a</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2025-06-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="exploration-across-scenarios" class="section level2">
<h2>Exploration across scenarios</h2>
<p>Modify <code>mean_effect</code> and <code>variance_effect</code> to
simulate different scenarios:</p>
<ul>
<li>SNP1 affects only the mean (<code>variance_effect = 0</code>).</li>
</ul>
<pre class="r"><code>simulated_data &lt;- simulate_vqtl_data(mean_effect = 0.1, variance_effect = 0)
trait_name &lt;- &quot;Y_vQTL&quot;
boxplot(phenotype[,trait_name] ~ genotype[,&quot;SNP1&quot;], data = simulated_data,
        xlab = &quot;Genotype&quot;,
        ylab = trait_name,
        col = c(&quot;#1f77b4&quot;, &quot;#ff7f0e&quot;, &quot;#2ca02c&quot;))</code></pre>
<p><img src="figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/joellembatchou/SISG2025_Association_Mapping/blob/5949e8a093c7e926391a8d3f70c194f137471340/docs/figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-14-1.png" target="_blank">5949e8a</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2025-06-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>qtl_pvals &lt;- lm_test(data = simulated_data, trait_name =  trait_name)
plot_qq(qtl_pvals)</code></pre>
<p><img src="figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-14-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-2">
Past versions of unnamed-chunk-14-2.png
</button>
</p>
<div id="fig-unnamed-chunk-14-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/joellembatchou/SISG2025_Association_Mapping/blob/5949e8a093c7e926391a8d3f70c194f137471340/docs/figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-14-2.png" target="_blank">5949e8a</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2025-06-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>levene_pvals &lt;- levene_test(data = simulated_data, trait_name =  trait_name)
plot_qq(levene_pvals)</code></pre>
<p><img src="figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-14-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-3">
Past versions of unnamed-chunk-14-3.png
</button>
</p>
<div id="fig-unnamed-chunk-14-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/joellembatchou/SISG2025_Association_Mapping/blob/5949e8a093c7e926391a8d3f70c194f137471340/docs/figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-14-3.png" target="_blank">5949e8a</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2025-06-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>gxe_pvals &lt;- gxe_test(data = simulated_data, trait_name =  trait_name)
plot_qq(gxe_pvals)</code></pre>
<p><img src="figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-14-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-4">
Past versions of unnamed-chunk-14-4.png
</button>
</p>
<div id="fig-unnamed-chunk-14-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/joellembatchou/SISG2025_Association_Mapping/blob/5949e8a093c7e926391a8d3f70c194f137471340/docs/figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-14-4.png" target="_blank">5949e8a</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2025-06-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li>SNP1 affects only the variance (<code>mean_effect = 0</code>).</li>
</ul>
<pre class="r"><code>simulated_data &lt;- simulate_vqtl_data(mean_effect = 0, variance_effect = 0.1)
trait_name &lt;- &quot;Y_vQTL&quot;
boxplot(phenotype[,trait_name] ~ genotype[,&quot;SNP1&quot;], data = simulated_data,
        xlab = &quot;Genotype&quot;,
        ylab = trait_name,
        col = c(&quot;#1f77b4&quot;, &quot;#ff7f0e&quot;, &quot;#2ca02c&quot;))</code></pre>
<p><img src="figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/joellembatchou/SISG2025_Association_Mapping/blob/5949e8a093c7e926391a8d3f70c194f137471340/docs/figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-15-1.png" target="_blank">5949e8a</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2025-06-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>qtl_pvals &lt;- lm_test(data = simulated_data, trait_name =  trait_name)
plot_qq(qtl_pvals)</code></pre>
<p><img src="figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-15-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-2">
Past versions of unnamed-chunk-15-2.png
</button>
</p>
<div id="fig-unnamed-chunk-15-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/joellembatchou/SISG2025_Association_Mapping/blob/5949e8a093c7e926391a8d3f70c194f137471340/docs/figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-15-2.png" target="_blank">5949e8a</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2025-06-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>levene_pvals &lt;- levene_test(data = simulated_data, trait_name =  trait_name)
plot_qq(levene_pvals)</code></pre>
<p><img src="figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-15-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-3">
Past versions of unnamed-chunk-15-3.png
</button>
</p>
<div id="fig-unnamed-chunk-15-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/joellembatchou/SISG2025_Association_Mapping/blob/5949e8a093c7e926391a8d3f70c194f137471340/docs/figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-15-3.png" target="_blank">5949e8a</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2025-06-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>gxe_pvals &lt;- gxe_test(data = simulated_data, trait_name =  trait_name)
plot_qq(gxe_pvals)</code></pre>
<p><img src="figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-15-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-4">
Past versions of unnamed-chunk-15-4.png
</button>
</p>
<div id="fig-unnamed-chunk-15-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/joellembatchou/SISG2025_Association_Mapping/blob/5949e8a093c7e926391a8d3f70c194f137471340/docs/figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-15-4.png" target="_blank">5949e8a</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2025-06-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li>SNP1 affects both the mean and variance.</li>
</ul>
<pre class="r"><code>simulated_data &lt;- simulate_vqtl_data(mean_effect = 0.1, variance_effect = 0.2)
trait_name &lt;- &quot;Y_vQTL&quot;
boxplot(phenotype[,trait_name] ~ genotype[,&quot;SNP1&quot;], data = simulated_data,
        xlab = &quot;Genotype&quot;,
        ylab = trait_name,
        col = c(&quot;#1f77b4&quot;, &quot;#ff7f0e&quot;, &quot;#2ca02c&quot;))</code></pre>
<p><img src="figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/joellembatchou/SISG2025_Association_Mapping/blob/5949e8a093c7e926391a8d3f70c194f137471340/docs/figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-16-1.png" target="_blank">5949e8a</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2025-06-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>qtl_pvals &lt;- lm_test(data = simulated_data, trait_name =  trait_name)
plot_qq(qtl_pvals)</code></pre>
<p><img src="figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-16-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-2">
Past versions of unnamed-chunk-16-2.png
</button>
</p>
<div id="fig-unnamed-chunk-16-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/joellembatchou/SISG2025_Association_Mapping/blob/5949e8a093c7e926391a8d3f70c194f137471340/docs/figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-16-2.png" target="_blank">5949e8a</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2025-06-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>levene_pvals &lt;- levene_test(data = simulated_data, trait_name =  trait_name)
plot_qq(levene_pvals)</code></pre>
<p><img src="figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-16-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-3">
Past versions of unnamed-chunk-16-3.png
</button>
</p>
<div id="fig-unnamed-chunk-16-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/joellembatchou/SISG2025_Association_Mapping/blob/5949e8a093c7e926391a8d3f70c194f137471340/docs/figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-16-3.png" target="_blank">5949e8a</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2025-06-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>gxe_pvals &lt;- gxe_test(data = simulated_data, trait_name =  trait_name)
plot_qq(gxe_pvals)</code></pre>
<p><img src="figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-16-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-4">
Past versions of unnamed-chunk-16-4.png
</button>
</p>
<div id="fig-unnamed-chunk-16-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/joellembatchou/SISG2025_Association_Mapping/blob/5949e8a093c7e926391a8d3f70c194f137471340/docs/figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-16-4.png" target="_blank">5949e8a</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2025-06-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>gxe_pvals &lt;- gxe_test(data = simulated_data, trait_name =  &quot;Y_gxe&quot;)
plot_qq(gxe_pvals)</code></pre>
<p><img src="figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-16-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-5">
Past versions of unnamed-chunk-16-5.png
</button>
</p>
<div id="fig-unnamed-chunk-16-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/joellembatchou/SISG2025_Association_Mapping/blob/5949e8a093c7e926391a8d3f70c194f137471340/docs/figure/QG3_Beyond_Standard_GWAS_practical_Key.Rmd/unnamed-chunk-16-5.png" target="_blank">5949e8a</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2025-06-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS 14.7.4

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/New_York
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] qqman_0.1.8

loaded via a namespace (and not attached):
 [1] jsonlite_1.8.5   compiler_4.3.0   highr_0.10       promises_1.2.0.1
 [5] Rcpp_1.0.10      stringr_1.5.0    git2r_0.32.0     later_1.3.1     
 [9] jquerylib_0.1.4  yaml_2.3.7       fastmap_1.1.1    R6_2.5.1        
[13] workflowr_1.7.0  knitr_1.43       MASS_7.3-58.4    tibble_3.2.1    
[17] rprojroot_2.0.3  bslib_0.5.0      pillar_1.9.0     rlang_1.1.1     
[21] utf8_1.2.3       cachem_1.0.8     calibrate_1.7.7  stringi_1.7.12  
[25] httpuv_1.6.11    xfun_0.39        fs_1.6.2         sass_0.4.6      
[29] cli_3.6.1        magrittr_2.0.3   digest_0.6.31    rstudioapi_0.14 
[33] lifecycle_1.0.3  vctrs_0.6.2      evaluate_0.21    glue_1.6.2      
[37] whisker_0.4.1    fansi_1.0.4      rmarkdown_2.22   tools_4.3.0     
[41] pkgconfig_2.0.3  htmltools_0.5.5 </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
