<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="" />


<title>Practical 1 - Association Testing</title>

<script src="site_libs/header-attrs-2.22/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SISG 2025 QG3</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Sessions
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Day 1</li>
    <li>
      <a href="Session01_practical.html">Session 1</a>
    </li>
    <li>
      <a href="Session02_practical.html">Session 2</a>
    </li>
    <li>
      <a href="Session03_practical.html">Session 3</a>
    </li>
    <li>
      <a href="SISGM15_prac4.html">Session 4</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Day 2</li>
    <li>
      <a href="SISGM15_prac5.html">Session 5</a>
    </li>
    <li>
      <a href="SISGM15_prac6.html">Session 6</a>
    </li>
    <li>
      <a href="Session07_practical.html">Session 7</a>
    </li>
    <li>
      <a href="Session08_practical.html">Session 8</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Day 3</li>
    <li>
      <a href="SISGM15_prac9.html">Session 9</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/joellembatchou/SISG2025_Association_Mapping">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Practical 1 - Association Testing</h1>
<h4 class="author"></h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-06-01
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>SISG2025_Association_Mapping/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230530code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230530)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230530code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230530)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjoellembatchouSISG2025AssociationMappingtreedaa49940eda83dcef08fc54029eb2ed3bab08d21targetblankdaa4994a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/joellembatchou/SISG2025_Association_Mapping/tree/daa49940eda83dcef08fc54029eb2ed3bab08d21" target="_blank">daa4994</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomjoellembatchouSISG2025AssociationMappingtreedaa49940eda83dcef08fc54029eb2ed3bab08d21targetblankdaa4994a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/joellembatchou/SISG2025_Association_Mapping/tree/daa49940eda83dcef08fc54029eb2ed3bab08d21" target="_blank">daa4994</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .qodo/
    Ignored:    analysis/.DS_Store
    Ignored:    data/run_regenie.r
    Ignored:    data/sim_rels_geno.bed
    Ignored:    exe/
    Ignored:    lectures/
    Ignored:    mk_website.R
    Ignored:    notes.txt
    Ignored:    tmp/

Untracked files:
    Untracked:  .mk_website.R.swp
    Untracked:  _workflowr.yml
    Untracked:  analysis/.index.Rmd.swp
    Untracked:  analysis/QG3_Association_Testing_practical_Key.Rmd
    Untracked:  analysis/QG3_Beyond_Standard_GWAS_practical_Key.Rmd
    Untracked:  analysis/QG3_CC_Imbalance_practical_Key.Rmd
    Untracked:  analysis/QG3_Downstream-Analyses_practical_Key.Rmd
    Untracked:  analysis/QG3_Plink_Population_Structure_practical_Key.Rmd
    Untracked:  analysis/QG3_Polygenic_Scores_practical_Key.Rmd
    Untracked:  analysis/QG3_Power-Design_practical_Key.Rmd
    Untracked:  analysis/QG3_RV_tests_practical_Key.Rmd
    Untracked:  analysis/QG3_Relatedness_REGENIE_practical_Key.Rmd

Unstaged changes:
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/QG3_Association_Testing_practical.Rmd</code>) and HTML
(<code>docs/QG3_Association_Testing_practical.html</code>) files. If
you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/joellembatchou/SISG2025_Association_Mapping/blob/daa49940eda83dcef08fc54029eb2ed3bab08d21/analysis/QG3_Association_Testing_practical.Rmd" target="_blank">daa4994</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2025-06-01
</td>
<td>
rename files
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Before you begin:</p>
<ul>
<li>Make sure that R is installed on your computer</li>
<li>For this lab, we will use a few R libraries:</li>
</ul>
<pre class="r"><code>library(data.table)</code></pre>
<div id="case-control-association-testing" class="section level2">
<h2>Case-Control Association Testing</h2>
<div id="introduction" class="section level3">
<h3>Introduction</h3>
<p>We will be using the <a
href="https://raw.githubusercontent.com/joellembatchou/SISG2025_Association_Mapping/master/data/LHON.txt">LHON
dataset</a> covered in the lecture notes for this portion of the
exercises. The LHON dataset is from a case-control study and includes
both phenotype and genotype data for a candidate gene.</p>
<p>Let’s first load the LHON data file into the R session. We need to
define the path to the file (if you have it downloaded on your machine,
change the path to the file location).</p>
<pre class="r"><code>LHON_FILE &lt;- &quot;https://raw.githubusercontent.com/joellembatchou/SISG2025_Association_Mapping/master/data/LHON.txt&quot; </code></pre>
<p>We can now read the file</p>
<pre class="r"><code>LHON &lt;- fread(LHON_FILE, header=TRUE)</code></pre>
</div>
<div id="helpful-suggestions-for-r" class="section level3">
<h3>Helpful suggestions for R</h3>
<p>There are many ways to obtain summary information for a dataset. Here
are some short examples:</p>
<ul>
<li>Get information on number of rows/columns as well as the variables
present in the data set</li>
</ul>
<pre class="r"><code>str(LHON)</code></pre>
<pre><code>Classes &#39;data.table&#39; and &#39;data.frame&#39;:  328 obs. of  3 variables:
 $ IID  : chr  &quot;ID1&quot; &quot;ID2&quot; &quot;ID3&quot; &quot;ID4&quot; ...
 $ GENO : chr  &quot;TT&quot; &quot;CT&quot; &quot;TT&quot; &quot;CT&quot; ...
 $ PHENO: chr  &quot;CONTROL&quot; &quot;CONTROL&quot; &quot;CASE&quot; &quot;CONTROL&quot; ...
 - attr(*, &quot;.internal.selfref&quot;)=&lt;externalptr&gt; </code></pre>
<ul>
<li>Get counts for a specific variable in the table (use <code>$</code>
to access a variable)</li>
</ul>
<pre class="r"><code>table(LHON$GENO)</code></pre>
<pre><code>
 CC  CT  TT 
 16  74 238 </code></pre>
<pre class="r"><code># cross tabulation for two variables
table(LHON$GENO, LHON$PHENO)</code></pre>
<pre><code>    
     CASE CONTROL
  CC    6      10
  CT    8      66
  TT   75     163</code></pre>
<ul>
<li>Functions like <code>as.numeric()</code> and <code>factor()</code>
will be useful to convert between numeric and categorical
variables.</li>
</ul>
<pre class="r"><code>LHON$GENO[1:5] # see the first 5 entries</code></pre>
<pre><code>[1] &quot;TT&quot; &quot;CT&quot; &quot;TT&quot; &quot;CT&quot; &quot;TT&quot;</code></pre>
<pre class="r"><code>as.numeric(factor(LHON$GENO, levels = c(&quot;CC&quot;, &quot;CT&quot;, &quot;TT&quot;)))[1:5] # convert to numeric specifying the order of the labels</code></pre>
<pre><code>[1] 3 2 3 2 3</code></pre>
<ul>
<li>Note: For any R function you don’t know the input syntax, you can
get that information using <code>?&lt;function_name&gt;</code>,
e.g. <code>?table</code></li>
</ul>
</div>
<div id="exercises" class="section level3">
<h3>Exercises</h3>
<p>Here are some things to look at:</p>
<ol style="list-style-type: decimal">
<li>Examine the variables in the dataset
<ul>
<li>How many observations? (use <code>str</code> function)</li>
<li>How many cases/controls? (use <code>table</code> function)</li>
<li>What are the genotypes present in the variable <code>GENO</code>?
(use <code>table</code> function)
<ul>
<li>To visualize the counts, you can use
<code>barplot(table(LHON$GENO))</code></li>
</ul></li>
<li>What is the distribution of the genotypes across cases/controls?
(use <code>table</code> function)</li>
</ul></li>
<li>Perform a logistic regression analysis for this data with
<code>CC</code> as the reference genotype using the <code>glm()</code>
function.</li>
</ol>
<ul>
<li>First convert the <code>GENO</code> variable to a factor</li>
</ul>
<pre class="r"><code>GENO_factor &lt;- factor(LHON$GENO, levels = c(&quot;CC&quot;, &quot;CT&quot;, &quot;TT&quot;)) # convert to numeric specifying the order of the labels</code></pre>
<ul>
<li>Convert the phenotype to a 0/1 variable</li>
</ul>
<pre class="r"><code>pheno_binary &lt;- 1 * (LHON$PHENO == &quot;CASE&quot;)</code></pre>
<ul>
<li>Check that the entries in <code>pheno_binary</code> with 1
correspond to <code>PHENO='CASE'</code></li>
<li>Run logistic regression using the <code>glm</code> function</li>
</ul>
<pre class="r"><code>logistic_model_LHON &lt;- glm(pheno_binary ~ GENO_factor, family = binomial(link = &quot;logit&quot;))</code></pre>
<p>You can get information about the model fit and parameter estimates
(i.e. coefficients):</p>
<pre class="r"><code>summary(logistic_model_LHON)</code></pre>
<ol start="3" style="list-style-type: decimal">
<li>Obtain odds ratios and confidence intervals (CI) for the
<code>CT</code> and <code>TT</code> genotypes relative to the
<code>CC</code> reference genotype. Interpret.
<ul>
<li>use the lecture notes to obtain odds ratios &amp; CI from estimates
and standard errors.</li>
</ul></li>
<li>Is there evidence of differences in odds of being a case for the
<code>CT</code> and <code>TT</code> genotypes (compared to
<code>CC</code>)?</li>
</ol>
<div id="extra" class="section level4">
<h4><em>Extra</em></h4>
<ol start="5" style="list-style-type: decimal">
<li>Perform the logistic regression analysis with an additive genotype
coding (e.g. counting the number of ‘T’ alleles).
<ul>
<li>Hint: To convert to numerical, create a new variable with values
0/1/2 based on the genotypes (you can then use <code>table()</code>
function to make sure the new variable was defined correctly).</li>
</ul></li>
</ol>
<pre class="r"><code>GENO_additive_T &lt;- 0 + 1 * (LHON$GENO == &quot;CT&quot;) + 2 * (LHON$GENO == &quot;TT&quot;)</code></pre>
<ol start="6" style="list-style-type: decimal">
<li>Obtain odds ratios and confidence intervals. Is there evidence of an
association? How does it compare with the 2-parameter model?</li>
</ol>
</div>
</div>
</div>
<div id="association-testing-with-quantitative-traits"
class="section level2">
<h2>Association Testing with Quantitative Traits</h2>
<div id="introduction-1" class="section level3">
<h3>Introduction</h3>
<p>We will be using the <a
href="https://raw.githubusercontent.com/joellembatchou/SISG2025_Association_Mapping/master/data/bpdata.csv">Blood
Pressure dataset</a> for this portion of the exercises. This dataset
contains diastolic and systolic blood pressure measurements for 1000
individuals, and genotype data at 11 SNPs in a candidate gene for blood
pressure. Covariates such as gender (sex) and body mass index (bmi) are
included as well.</p>
<p>Let’s first load the file into R. We need to define the path to the
file (if you have it downloaded on your machine, change the path to the
file location).</p>
<pre class="r"><code>BP_FILE &lt;- &quot;https://raw.githubusercontent.com/joellembatchou/SISG2025_Association_Mapping/master/data/bpdata.csv&quot; </code></pre>
<p>Use the following command to read it into R:</p>
<pre class="r"><code>BP &lt;- fread(BP_FILE, header=TRUE)</code></pre>
<ul>
<li>Get a snippet of the data:</li>
</ul>
<pre class="r"><code>head(BP, 2)</code></pre>
<pre><code>   V1    sex sbp dbp snp1 snp2 snp3 snp4 snp5 snp6 snp7 snp8 snp9 snp10 snp11
1:  1 FEMALE 171  89   CC   TT   TT   TT   CC   GG   AA   TT   TT    CC    TT
2:  2   MALE 160  99   TT   TT   CC &lt;NA&gt;   CC   AG   AT   CC   CT    CC    CT
   bmi
1:  25
2:  35</code></pre>
</div>
<div id="exercises-1" class="section level3">
<h3>Exercises</h3>
<p>Here are some things to try:</p>
<ol style="list-style-type: decimal">
<li>Perform a linear regression of systolic blood pressure
(<code>sbp</code>) on <code>SNP3</code> using the <code>lm()</code>
function.</li>
</ol>
<pre class="r"><code>linear_model_BP &lt;- lm(sbp ~ snp3, data = BP)</code></pre>
<p>You can get information about the model fit and parameter estimates
(i.e. coefficients):</p>
<pre class="r"><code>summary(linear_model_BP)</code></pre>
<ol start="2" style="list-style-type: decimal">
<li><p>Is there any evidence of an effect of the SNP on systolic blood
pressure?</p></li>
<li><p>Provide a plot illustrating the relationship between sbp and the
three genotypes at SNP3.</p></li>
</ol>
<pre class="r"><code>with(BP, boxplot(sbp ~ snp3))</code></pre>
<ul>
<li>How does it compare with the linear model fitted in question
(1)?</li>
</ul>
<ol start="4" style="list-style-type: decimal">
<li><p>By default, the 2-parameter model is used since the SNP is stored
in the data as categorical. Contrast the parameter estimates, p-values
and confidence intervals obtained between this model and using:</p>
<ul>
<li>additive (linear) model (counting the T allele)</li>
<li>dominant model</li>
<li>recessive model</li>
</ul></li>
</ol>
<p><em>Hint</em>: for each case, generate the appropriate allele coding
variable and pass it to the <code>lm()</code> function. For example with
additive coding:</p>
<pre class="r"><code>SNP3_additive &lt;- 0 + 1 * (BP$snp3 == &quot;TC&quot;) + 2 * (BP$snp3 == &quot;TT&quot;)
linear_model_BP_additive &lt;- lm(sbp ~ SNP3_additive, data = BP)</code></pre>
<p>For question 5 and 6 below, R also has a ‘formula’ syntax, frequently
used when specifying regression models with many predictors. To regress
an outcome <code>y</code> on several covariates, the syntax is:</p>
<pre class="r"><code>lm(y ~ covariate1 + covariate2 + covariate3)</code></pre>
<ol start="5" style="list-style-type: decimal">
<li>Now redo the linear regression analysis of <code>sbp</code> from
question 4 for the <strong>additive model</strong>, but this time adjust
for <code>sex</code> and <code>bmi</code>. Do the results change?</li>
</ol>
<div id="extra-1" class="section level4">
<h4><em>Extra</em></h4>
<ol start="6" style="list-style-type: decimal">
<li>What proportion of the variance of <code>sbp</code> is explained by
all 11 SNPs combined using categorical coding?
<ul>
<li>Use the <code>summary()</code> function to see the model results
(the proportion of variance is the “Multiple R-squared” quantity)</li>
<li>How would it differ if an additive coding is used for the 11 SNPs?
<ul>
<li>use <code>unique()</code> to check the genotypes for each SNP,
e.g. <code>unique(BP$snp1)</code></li>
<li>count the T allele (or A allele if applicable)</li>
</ul></li>
</ul></li>
</ol>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS 14.7.4

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] data.table_1.14.8

loaded via a namespace (and not attached):
 [1] vctrs_0.6.2      cli_3.6.1        knitr_1.43       rlang_1.1.1     
 [5] xfun_0.39        stringi_1.7.12   promises_1.2.0.1 jsonlite_1.8.5  
 [9] workflowr_1.7.0  glue_1.6.2       rprojroot_2.0.3  git2r_0.32.0    
[13] htmltools_0.5.5  httpuv_1.6.11    sass_0.4.6       fansi_1.0.4     
[17] rmarkdown_2.22   evaluate_0.21    jquerylib_0.1.4  tibble_3.2.1    
[21] fastmap_1.1.1    yaml_2.3.7       lifecycle_1.0.3  whisker_0.4.1   
[25] stringr_1.5.0    compiler_4.3.0   fs_1.6.2         Rcpp_1.0.10     
[29] pkgconfig_2.0.3  rstudioapi_0.14  later_1.3.1      digest_0.6.31   
[33] R6_2.5.1         utf8_1.2.3       curl_5.0.1       pillar_1.9.0    
[37] magrittr_2.0.3   bslib_0.5.0      tools_4.3.0      cachem_1.0.8    </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
