<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="" />


<title>Practical 2 Key - Population Structure Inference</title>

<script src="site_libs/header-attrs-2.22/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SISG 2025 QG3</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Sessions
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Day 1</li>
    <li>
      <a href="QG3_Association_Testing_practical.html">Session 1</a>
    </li>
    <li>
      <a href="QG3_Plink_Population_Structure_practical.html">Session 2</a>
    </li>
    <li>
      <a href="QG3_Relatedness_REGENIE_practical.html">Session 3</a>
    </li>
    <li>
      <a href="QG3_RV_tests_practical.html">Session 4</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Day 2</li>
    <li>
      <a href="QG3_Beyond_Standard_GWAS_practical.html">Session 5</a>
    </li>
    <li>
      <a href="QG3_Downstream-Analyses_practical.html">Session 6</a>
    </li>
    <li>
      <a href="QG3_Polygenic_Scores_practical.html">Session 7</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Day 3</li>
    <li>
      <a href="QG3_Power-Design_practical.html">Session 9</a>
    </li>
    <li>
      <a href="QG3_CC_Imbalance_practical_Key.html">Session 10</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/joellembatchou/SISG2025_Association_Mapping">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Practical 2 Key - Population Structure
Inference</h1>
<h4 class="author"></h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-06-01
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>SISG2025_Association_Mapping/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230530code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230530)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230530code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230530)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjoellembatchouSISG2025AssociationMappingtree421fcd0acf1b40a567c813b930acd5b9eb5bb768targetblank421fcd0a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/joellembatchou/SISG2025_Association_Mapping/tree/421fcd0acf1b40a567c813b930acd5b9eb5bb768" target="_blank">421fcd0</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomjoellembatchouSISG2025AssociationMappingtree421fcd0acf1b40a567c813b930acd5b9eb5bb768targetblank421fcd0a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/joellembatchou/SISG2025_Association_Mapping/tree/421fcd0acf1b40a567c813b930acd5b9eb5bb768" target="_blank">421fcd0</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .qodo/
    Ignored:    analysis/.DS_Store
    Ignored:    data/run_regenie.r
    Ignored:    data/sim_rels_geno.bed
    Ignored:    exe/
    Ignored:    ldRef.log
    Ignored:    lectures/
    Ignored:    mk_website.R
    Ignored:    notes.txt
    Ignored:    test1.log
    Ignored:    test2.log
    Ignored:    test3.log
    Ignored:    test4.log
    Ignored:    tmp/

Untracked files:
    Untracked:  .mk_website.R.swp
    Untracked:  GWAS.ma
    Untracked:  _workflowr.yml
    Untracked:  analysis/.index.Rmd.swp
    Untracked:  analysis/QG3_Association_Testing_practical_Key.Rmd
    Untracked:  analysis/QG3_Beyond_Standard_GWAS_practical_Key.Rmd
    Untracked:  analysis/QG3_CC_Imbalance_practical_Key.Rmd
    Untracked:  analysis/QG3_Downstream-Analyses_practical_Key.Rmd
    Untracked:  analysis/QG3_Plink_Population_Structure_practical_Key.Rmd
    Untracked:  analysis/QG3_Polygenic_Scores_practical_Key.Rmd
    Untracked:  analysis/QG3_Power-Design_practical_Key.Rmd
    Untracked:  analysis/QG3_RV_tests_practical_Key.Rmd
    Untracked:  analysis/QG3_Relatedness_REGENIE_practical_Key.Rmd
    Untracked:  causals.snplist
    Untracked:  ldRef.bed
    Untracked:  ldRef.bim
    Untracked:  ldRef.fam
    Untracked:  ldRef.map
    Untracked:  ldRef.ped
    Untracked:  sim.config
    Untracked:  test1.cma.cojo
    Untracked:  test1.jma.cojo
    Untracked:  test1.ldr.cojo
    Untracked:  test2.cma.cojo
    Untracked:  test2.jma.cojo
    Untracked:  test2.ldr.cojo
    Untracked:  test3.cma.cojo
    Untracked:  test3.jma.cojo
    Untracked:  test3.ldr.cojo
    Untracked:  test4.cma.cojo
    Untracked:  test4.jma.cojo
    Untracked:  test4.ldr.cojo

Unstaged changes:
    Modified:   analysis/QG3_Downstream-Analyses_practical.Rmd
    Modified:   analysis/_site.yml

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<p>Before you begin:</p>
<ul>
<li>Make sure that R is installed on your computer</li>
<li>For this lab, we will use the following R libraries:</li>
</ul>
<pre class="r"><code>library(data.table)
library(dplyr)
library(bigsnpr)
library(ggplot2)</code></pre>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>We will be working with a subset of the genotype data from the Human
Genome Diversity Panel (HGDP) and HapMap.</p>
<p>The file “<a
href="https://github.com/joellembatchou/SISG2024_Association_Mapping/tree/master/data">YRI_CEU_ASW_MEX_NAM.bed</a>”
is a binary file in PLINK BED format with accompanying BIM and FAM
files. It contains the genotype data at autosomal SNPs (i.e. chromosomes
1-22) for:</p>
<ul>
<li>Native American samples from HGDP</li>
<li>Four population samples from HapMap:
<ul>
<li>Yoruba in Ibadan, Nigeria (YRI)</li>
<li>Utah residents with ancestry from Northern and Western Europe
(CEU)</li>
<li>Mexican Americans in Los Angeles, California (MXL)</li>
<li>African Americans from the south-western United States (ASW)</li>
</ul></li>
</ul>
<p><strong>File with ancestry labels assignment for each
sample</strong>: <a
href="https://raw.githubusercontent.com/joellembatchou/SISG2024_Association_Mapping/master/data/Population_Sample_Info.txt">Population_Sample_Info.txt</a></p>
<div id="data-preparation" class="section level3">
<h3>Data preparation</h3>
<p>Let’s first load the HGDP data into the R session. We need to define
the path to the directory containing the PLINK BED and the ancestry
label files (change the path to the file location).</p>
<pre class="r"><code># change this to the directory on your machine
HGDP_dir &lt;- &quot;/SISGM19/data/&quot; </code></pre>
<p>Also specify the path to the PLINK2 binary</p>
<pre class="r"><code>plink2_binary &lt;- &quot;/SISGM19/bin/plink2&quot; </code></pre>
<p>We can now read the PLINK BED and FAM files (recall the BED file is a
binary file):</p>
<pre class="r"><code>HGDP_bim &lt;- fread(sprintf(&quot;%s/YRI_CEU_ASW_MEX_NAM.bim&quot;, HGDP_dir), header = FALSE)
head(HGDP_bim, 3)</code></pre>
<pre><code>   V1        V2 V3      V4 V5 V6
1:  1 rs9442372  0 1008567  1  2
2:  1 rs2887286  0 1145994  1  2
3:  1 rs3813199  0 1148140  1  2</code></pre>
<pre class="r"><code>HGDP_fam &lt;- fread(sprintf(&quot;%s/YRI_CEU_ASW_MEX_NAM.fam&quot;, HGDP_dir), header = FALSE)
head(HGDP_fam, 3)</code></pre>
<pre><code>     V1        V2 V3 V4 V5 V6
1: 1432 HGDP00702  0  0  2 -9
2: 1433 HGDP00703  0  0  1 -9
3: 1434 HGDP00704  0  0  2 -9</code></pre>
<p>When reading the ancestry label file, we need to make sure the order
of samples matches that in the PLINK data:</p>
<pre class="r"><code>HGDP_ancestry_df &lt;- fread(sprintf(&quot;%s/Population_Sample_Info.txt&quot;, HGDP_dir))
HGDP_ancestry_df &lt;- left_join(HGDP_fam[,c(&quot;V1&quot;,&quot;V2&quot;)], HGDP_ancestry_df, by = c(&quot;V1&quot; = &quot;FID&quot;, &quot;V2&quot; = &quot;IID&quot;))
head(HGDP_ancestry_df, 3)</code></pre>
<pre><code>     V1        V2 Population
1: 1432 HGDP00702        NAM
2: 1433 HGDP00703        NAM
3: 1434 HGDP00704        NAM</code></pre>
</div>
</div>
<div id="exercises" class="section level2">
<h2>Exercises</h2>
<p>Here are some things to look at:</p>
<ol style="list-style-type: decimal">
<li>Examine the dataset:</li>
</ol>
<ul>
<li>How many samples are present?</li>
</ul>
<pre class="r"><code>str(HGDP_fam)</code></pre>
<pre><code>Classes &#39;data.table&#39; and &#39;data.frame&#39;:  604 obs. of  6 variables:
 $ V1: chr  &quot;1432&quot; &quot;1433&quot; &quot;1434&quot; &quot;1436&quot; ...
 $ V2: chr  &quot;HGDP00702&quot; &quot;HGDP00703&quot; &quot;HGDP00704&quot; &quot;HGDP00706&quot; ...
 $ V3: chr  &quot;0&quot; &quot;0&quot; &quot;0&quot; &quot;0&quot; ...
 $ V4: chr  &quot;0&quot; &quot;0&quot; &quot;0&quot; &quot;0&quot; ...
 $ V5: int  2 1 2 2 2 1 2 1 2 1 ...
 $ V6: int  -9 -9 -9 -9 -9 -9 -9 -9 -9 -9 ...
 - attr(*, &quot;.internal.selfref&quot;)=&lt;externalptr&gt; </code></pre>
<ul>
<li>How many SNPs?</li>
</ul>
<pre class="r"><code>str(HGDP_bim)</code></pre>
<pre><code>Classes &#39;data.table&#39; and &#39;data.frame&#39;:  150872 obs. of  6 variables:
 $ V1: int  1 1 1 1 1 1 1 1 1 1 ...
 $ V2: chr  &quot;rs9442372&quot; &quot;rs2887286&quot; &quot;rs3813199&quot; &quot;rs6685064&quot; ...
 $ V3: int  0 0 0 0 0 0 0 0 0 0 ...
 $ V4: int  1008567 1145994 1148140 1201155 1452629 1878053 2013924 2023116 2072349 2072426 ...
 $ V5: int  1 1 1 1 1 1 1 1 1 1 ...
 $ V6: int  2 2 2 2 2 2 2 2 2 2 ...
 - attr(*, &quot;.internal.selfref&quot;)=&lt;externalptr&gt; </code></pre>
<ul>
<li>What is the number of samples in each population?</li>
</ul>
<pre class="r"><code>table(HGDP_ancestry_df$Population)</code></pre>
<pre><code>
ASW CEU MXL NAM YRI 
 87 165  86  63 203 </code></pre>
<ol start="2" style="list-style-type: decimal">
<li>Get the first 10 principal components (PCs) in PLINK using all
SNPs.</li>
</ol>
<pre class="r"><code>cmd &lt;- sprintf(&quot;%s --bfile %s/YRI_CEU_ASW_MEX_NAM --pca 10 --out pca_plink&quot;, plink2_binary, HGDP_dir)
system(cmd)</code></pre>
<p>This generates two files <code>pca_plink.eigenvec</code> containing
the PCs (eigenvectors), and <code>pca_plink.eigenval</code> containing
the top eigenvalues.</p>
<ol start="3" style="list-style-type: decimal">
<li>Read in the PCs in R and make a scatterplot of the first two PCs
with each point colored by population membership.</li>
</ol>
<pre class="r"><code>PC_df &lt;- left_join(HGDP_ancestry_df, fread(&quot;pca_plink.eigenvec&quot;), by = c(&quot;V1&quot; = &quot;#FID&quot;, &quot;V2&quot; = &quot;IID&quot;))
ggplot(PC_df, aes(x=PC1, y=PC2, color = Population)) +
  geom_point()</code></pre>
<p><img src="figure/QG3_Plink_Population_Structure_practical_Key.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<ol start="4" style="list-style-type: decimal">
<li><p>Interpret the first two PCs, what ancestries are they
reflecting?</p></li>
<li><p>Read in the eigenvalues and make a scree plot corresponding for
these first 10 PCs. Estimate the proportion of variance explained by the
first two PCs.</p></li>
</ol>
<pre class="r"><code>eigenvalues_df &lt;- fread(&quot;pca_plink.eigenval&quot;, header = FALSE)
ggplot(eigenvalues_df, aes(x = 1:10, y = V1)) +
  geom_point() +
  geom_line() +
  scale_x_continuous(breaks = 1:10) +
  labs(x = &quot;PC&quot;, y = &quot;Eigenvalue&quot;)</code></pre>
<p><img src="figure/QG3_Plink_Population_Structure_practical_Key.Rmd/eigen-vals-comp-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>sum(eigenvalues_df$V1[1:2]) / sum(eigenvalues_df$V1)</code></pre>
<pre><code>[1] 0.8308752</code></pre>
<ol start="6" style="list-style-type: decimal">
<li>Now redo Question 2 above using the <a
href="https://privefl.github.io/bigsnpr/reference/index.html"><code>bigsnpr</code>
R package</a> specifying a <span class="math inline">\(r^2\)</span>
threshold of 0.2 (i.e. LD pruning) as well as a minimum minor allele
count (MAC) of 20.</li>
</ol>
<pre class="r"><code>obj.bed &lt;- bed(bedfile = sprintf(&quot;%s/YRI_CEU_ASW_MEX_NAM.bed&quot;, HGDP_dir))
pca.bigsnpr &lt;- bed_autoSVD(
  obj.bed, 
  thr.r2 = 0.2, # R^2 threshold
  k = 10, # number of PCs
  min.mac = 20 # minimum minor allele count (MAC) filter
)</code></pre>
<pre><code>
Phase of clumping (on MAC) at r^2 &gt; 0.2.. keep 87127 variants.
Discarding 48 variants with MAC &lt; 20.

Iteration 1:
Computing SVD..</code></pre>
<pre><code>The default of &#39;doScale&#39; is FALSE now for stability;
  set options(mc_doScale_quiet=TRUE) to suppress this (once per session) message</code></pre>
<pre><code>0 outlier variant detected..

Converged!</code></pre>
<ul>
<li>You can evaluate the PCA results using</li>
</ul>
<pre class="r"><code># plot PC2 vs PC1
plot(pca.bigsnpr, type = &quot;scores&quot;, scores = 1:2)</code></pre>
<p><img src="figure/QG3_Plink_Population_Structure_practical_Key.Rmd/bigsnpr-evals-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># scree plot
plot(pca.bigsnpr) </code></pre>
<p><img src="figure/QG3_Plink_Population_Structure_practical_Key.Rmd/bigsnpr-evals-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># plot SNP loadings for the first two PCs
plot(pca.bigsnpr, type = &quot;loadings&quot;, loadings = 1:2, coeff = 0.4)</code></pre>
<p><img src="figure/QG3_Plink_Population_Structure_practical_Key.Rmd/bigsnpr-evals-3.png" width="672" style="display: block; margin: auto;" /></p>
<ul>
<li>Make a scatter plot of the first two principal components (PCs) with
each point colored according to population membership. Does the plot
change from the one in Question 2?</li>
</ul>
<pre class="r"><code>plot(pca.bigsnpr, type = &quot;scores&quot;, scores = 1:2) +
  aes(color = HGDP_ancestry_df$Population) +
  labs(color = &quot;Population&quot;)</code></pre>
<p><img src="figure/QG3_Plink_Population_Structure_practical_Key.Rmd/plot-pca-bigsnpr-1.png" width="672" style="display: block; margin: auto;" /></p>
<ul>
<li>Check the SNP loadings for the first 5 PCs.</li>
</ul>
<pre class="r"><code>plot(pca.bigsnpr, type = &quot;loadings&quot;, loadings = 1:5, coeff = 0.4)</code></pre>
<p><img src="figure/QG3_Plink_Population_Structure_practical_Key.Rmd/pc-five-loadings-1.png" width="672" style="display: block; margin: auto;" /></p>
<ol start="7" style="list-style-type: decimal">
<li>Predict the proportion of Native American ancestry for the HapMap
MXL based on the PCA output in Question 3 <em>using one of the principal
components</em> and assuming that the HapMap MXL have negligible African
Ancestry. Which PC is most appropriate for this analysis?</li>
</ol>
<p><em>Hint</em>: To compute the average PC2 value for individuals of
CEU ancestry</p>
<pre class="r"><code>ceu.mean &lt;- with(PC_df, mean(PC2[Population == &quot;CEU&quot;]))</code></pre>
<p>Do the same for individuals of NAM ancestry. How can you express
distances of the MXL individuals relative to those means based on the
chosen PC?</p>
<pre class="r"><code>nam.mean &lt;- with(PC_df, mean(PC2[Population == &quot;NAM&quot;]))
mxl.prop.nam &lt;- with(PC_df, (ceu.mean - PC2[Population == &quot;MXL&quot;]) / abs(ceu.mean - nam.mean))
summary(mxl.prop.nam)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
0.05325 0.37687 0.48112 0.47207 0.57836 0.81147 </code></pre>
<ol start="8" style="list-style-type: decimal">
<li>Make a barplot of the proportional Native American/European ancestry
estimates based on your output of question 7.</li>
</ol>
<pre class="r"><code>sorted.props.nam &lt;- sort(mxl.prop.nam, decreasing = TRUE)
df_mxl_props &lt;- data.frame(
  anc.props = c(sorted.props.nam, 1 - sorted.props.nam),
  x = rep(1:length(sorted.props.nam), times = 2),
  population.labels = rep(c(&quot;NAM&quot;, &quot;CEU&quot;), each = length(sorted.props.nam))
)
ggplot(df_mxl_props, aes(x = x, y = anc.props, fill = population.labels)) +
  geom_bar(position=&quot;stack&quot;, stat=&quot;identity&quot;) +
  labs(x=&quot;Sample&quot;, y = &quot;Ancestry Proportion&quot;, fill = &quot;Population&quot;)</code></pre>
<p><img src="figure/QG3_Plink_Population_Structure_practical_Key.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="extra" class="section level3">
<h3><em>Extra</em></h3>
<ol start="9" style="list-style-type: decimal">
<li>Check if there are samples related 2nd degree or closer. If so, run
PCA as in Question 6 removing these samples then project the remaining
samples onto the PC space. Plot the top 2 PCs.</li>
</ol>
<ul>
<li>The command to check relatedness is:</li>
</ul>
<pre class="r"><code># check for 2nd degree relateds or closer
relatedness_info &lt;- snp_plinkKINGQC(
  plink2.path = plink2_binary, 
  bedfile.in = sprintf(&quot;%s/YRI_CEU_ASW_MEX_NAM.bed&quot;, HGDP_dir), 
  thr.king = 2^-3.5, # threshold to identify 2nd degree relateds
  make.bed = FALSE
)</code></pre>
<p>This returns a data frame which contains all pairs of individuals
related 2nd degree or closer.</p>
<pre class="r"><code>str(relatedness_info)</code></pre>
<pre><code>&#39;data.frame&#39;:   362 obs. of  8 variables:
 $ FID1   : chr  &quot;1563&quot; &quot;1567&quot; &quot;1567&quot; &quot;1570&quot; ...
 $ IID1   : chr  &quot;HGDP00845&quot; &quot;HGDP00849&quot; &quot;HGDP00849&quot; &quot;HGDP00852&quot; ...
 $ FID2   : chr  &quot;1556&quot; &quot;1556&quot; &quot;1561&quot; &quot;1551&quot; ...
 $ IID2   : chr  &quot;HGDP00838&quot; &quot;HGDP00838&quot; &quot;HGDP00843&quot; &quot;HGDP00832&quot; ...
 $ NSNP   : int  150801 150802 150832 150816 150822 150814 150819 150815 150674 150796 ...
 $ HETHET : num  0.0976 0.1069 0.1059 0.099 0.1022 ...
 $ IBS0   : num  0.0221 0.0219 0.0223 0.0219 0.0227 ...
 $ KINSHIP: num  0.104 0.126 0.13 0.118 0.118 ...</code></pre>
<ul>
<li>We can then remove them when calling <code>bed_autoSVD()</code>
using the <code>ind.row</code> argument.</li>
</ul>
<pre class="r"><code>ind.rel &lt;- match(c(relatedness_info$IID1, relatedness_info$IID2), obj.bed$fam$sample.ID) # relateds
ind.norel &lt;- rows_along(obj.bed)[-ind.rel] # unrelateds
# Run PCA on unrelateds
obj.svd2 &lt;- bed_autoSVD(
  obj.bed, 
  thr.r2 = 0.2, # R^2 threshold
  k = 10, # number of PCs
  min.mac = 20, # minimum minor allele count (MAC) filter
  ind.row = ind.norel
)</code></pre>
<pre><code>
Phase of clumping (on MAC) at r^2 &gt; 0.2.. keep 77173 variants.
Discarding 12226 variants with MAC &lt; 20.

Iteration 1:
Computing SVD..
0 outlier variant detected..

Converged!</code></pre>
<ul>
<li>Use <code>bed_projectSelfPCA()</code> to project related samples on
the PC space. <em>(Hint: This <a
href="https://privefl.github.io/bigsnpr/articles/bedpca.html">tutorial
document</a> from <code>bigsnpr</code> will be helpful – see the last
section ‘Project remaining individuals’)</em></li>
</ul>
<pre class="r"><code>PCs &lt;- matrix(NA, nrow(obj.bed), ncol(obj.svd2$u))
# PCs for unrelateds
PCs[ind.norel, ] &lt;- predict(obj.svd2)
# Project relateds on PC space
proj &lt;- bed_projectSelfPCA(
  obj.svd2, 
  obj.bed,
  ind.row = ind.rel
  )
PCs[ind.rel, ] &lt;- proj$OADP_proj

# Plot the top 2 PCs with projections
pop_palette &lt;- c(&quot;#E69F00&quot;, &quot;#56B4E9&quot;, &quot;#009E73&quot;, &quot;#F0E442&quot;, &quot;#0072B2&quot;)
names(pop_palette) &lt;- unique(HGDP_ancestry_df$Population)

plot( # for unrelateds
  PCs[ind.norel, 1:2], 
  col = pop_palette[HGDP_ancestry_df$Population[ind.norel]], 
  pch = 1, xlab = &quot;PC1&quot;, ylab = &quot;PC2&quot;
  )
points( # for relateds
  PCs[ind.rel, 1:2], 
  col = pop_palette[HGDP_ancestry_df$Population[ind.rel]], 
  pch = 2
  )
# add the legends
legend(&quot;topleft&quot;, legend = names(pop_palette), col = pop_palette, pch = 19, title = &quot;Population&quot;)
legend(&quot;topright&quot;, legend = c(&quot;Model&quot;, &quot;Projected&quot;), col = c(&quot;black&quot;, &quot;black&quot;), pch = c(1, 2))</code></pre>
<p><img src="figure/QG3_Plink_Population_Structure_practical_Key.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS 14.7.4

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggplot2_3.4.2     bigsnpr_1.12.9    bigstatsr_1.5.12  dplyr_1.1.2      
[5] data.table_1.14.8

loaded via a namespace (and not attached):
 [1] gtable_0.3.3       xfun_0.39          bslib_0.5.0        lattice_0.21-8    
 [5] bigassertr_0.1.6   vctrs_0.6.2        tools_4.3.0        ps_1.7.5          
 [9] generics_0.1.3     parallel_4.3.0     tibble_3.2.1       fansi_1.0.4       
[13] DEoptimR_1.0-14    highr_0.10         pkgconfig_2.0.3    Matrix_1.5-4      
[17] rngtools_1.5.2     lifecycle_1.0.3    compiler_4.3.0     farver_2.1.1      
[21] stringr_1.5.0      git2r_0.32.0       munsell_0.5.0      bigparallelr_0.3.2
[25] codetools_0.2-19   httpuv_1.6.11      htmltools_0.5.5    sass_0.4.6        
[29] yaml_2.3.7         hexbin_1.28.3      later_1.3.1        pillar_1.9.0      
[33] jquerylib_0.1.4    cachem_1.0.8       doRNG_1.8.6        iterators_1.0.14  
[37] foreach_1.5.2      robustbase_0.99-0  parallelly_1.36.0  RSpectra_0.16-1   
[41] tidyselect_1.2.0   digest_0.6.31      stringi_1.7.12     labeling_0.4.2    
[45] cowplot_1.1.1      rprojroot_2.0.3    fastmap_1.1.1      grid_4.3.0        
[49] colorspace_2.1-0   cli_3.6.1          magrittr_2.0.3     utf8_1.2.3        
[53] bigutilsr_0.3.4    withr_2.5.0        scales_1.2.1       promises_1.2.0.1  
[57] rmarkdown_2.22     bigsparser_0.6.1   rmio_0.4.0         bit_4.0.5         
[61] bigreadr_0.2.5     workflowr_1.7.0    evaluate_0.21      knitr_1.43        
[65] ff_4.0.9           doParallel_1.0.17  viridisLite_0.4.2  rlang_1.1.1       
[69] Rcpp_1.0.10        glue_1.6.2         rstudioapi_0.14    jsonlite_1.8.5    
[73] R6_2.5.1           fs_1.6.2           flock_0.7         </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
