---
title: "Practical 8 - Case-control imbalance in GWAS"
author: ""
date: ""
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---
Before you begin:

* Make sure that R is installed on your computer
* For this lab, we will use the following R libraries:
```{r load-libs, message=FALSE, warning=FALSE}
library(data.table)
library(dplyr)
library(qqman)
library(ggplot2)
```

## Introduction
We will generate a simulated dataset consisting of 3 binary traits with different amounts of case-control imbalance, and a genetic data set of **null** SNPs to examine the distribution of the test statistics when  scanning for associations.

## Data preparation
We first need to define the path to the PLINK 1.9 and REGENIE binaries.
```{r echo=FALSE}
plink_binary <- "/Users/joelle.mbatchou/software/bins/plink1.9" 
regenie_binary <- "/Users/joelle.mbatchou/software/bins/regenie"
```

```{r eval=FALSE}
plink_binary <- "/SISGM19/bin/plink1.9" 
regenie_binary <- "/SISGM19/bin/regenie" 
```

If you don't have REGENIE installed on your machine, download the R implementation
[here](https://gatech.box.com/s/k7yk3tpqlekuc7zh6xquyjh9vbvumcx7) and change the path of the variable `regenie_script` to the path of the script on your machine.
```{r echo = FALSE}
regenie_script <- "data/run_regenie.r"
source(regenie_script)
```

```{r eval = FALSE}
regenie_script <- "/SISGM19/data/run_regenie.r"
source(regenie_script)
```



## Simulate the data
We use PLINK1.9 to simulate the genetic dataset. For $N=10,000$ samples, let's simulate 10,000 variants where 5,000 are common with MAF chosen from a Uniform(0.05, 0.5) distribution and for the rare variants, we will use a Uniform(0.001, 0.01) distribution. Run the following command in R to get the simulated data:
```{r, eval = FALSE}
N <- 10e3
# Generate a configuration file specifying allele frequencies (a,b) for Uniform(a,b) distribution
write(paste0("5000 common 0.05 0.5 1 1"), "sim.config")
write(paste0("5000 rare 0.001 0.01 1 1"), "sim.config", append = TRUE)
# Run PLINK1.9
cmd <- sprintf("%s --make-bed --simulate sim.config --simulate-ncases %d --simulate-ncontrols 0 --simulate-prevalence 0.1  --out cc_imb_geno", plink_binary, N)
system(cmd, intern = T)
```
You should now have files `cc_imb_geno.{bed,bim,fam}`.

For the phenotype data simulation, we will simulate 3 phenotypes with different levels of case-control imbalance (casse-control ratios [CCR] 1:9, 1:99, and 1:199). Run the following code
```{r, eval = FALSE}
# get FID/IID from FAM file
sample.ids <- fread("cc_imb_geno.fam", header = FALSE)
N <- nrow(sample.ids)

## Set prevalence = 10% (CCR 1:9)
y1 <- rbinom(N, 1, prob = 0.1 )
## Set prevalence = 1% (CCR 1:99)
y2 <- rbinom(N, 1, prob = 0.01 )
## Set prevalence = 0.5% (CCR 1:199)
y3 <- rbinom(N, 1, prob = 0.005 )

# write to file
fwrite(
  data.frame(FID = sample.ids$V1, IID = sample.ids$V2, Y1 = y1, Y2 = y2, Y3 = y3),
  "cc_imb_pheno.txt", 
  sep = "\t", na = NA, quote = FALSE
)
```
You should now have a file named `cc_imb_pheno.txt`.

## Exercises
We will now assess the null distribution of our test statistics when performing association testing using different models.

1.  Run the GWAS in REGENIE for the 3 traits.
```{r, eval = FALSE}
cmd <- sprintf('%s --bed cc_imb_geno --phenoFile cc_imb_pheno.txt --bt --step 2 --bsize 2000 --ignore-pred --out test_regenie', regenie_binary)
system(cmd, intern = T)
```
This will produce three sumstats files (one for each phenotype) which you can read in R:
```{r eval=FALSE}
sumstats.y1 <- fread("test_regenie_Y1.regenie") 
```


**<span style="color: red;"> To run the R implementation instead</span>**, run the following for each trait (this computes the association tests and stores it in the R variable directly)
```{r eval=FALSE}
sumstats.y1 <- run_regenie_step2_bt(
  bedfile = "cc_imb_geno",
  phenofile = "cc_imb_pheno.txt",
  phenocol = "Y1",
  bsize = 300
) 
```

2. Make a QQ plot of the p-values for each phenotype. Since these are null SNPs, how does it compare to what we expect? 
```{r eval=FALSE}
qq(10^-sumstats.y1$LOG10P)
```

3. Make a histogram of the test statistics for each phenotype and overlay with a normal distribution. How well do they match? We will create a R function to easily make this plot for different phenotypes. 
```{r, eval = FALSE}
plot.sumstats.hist <- function(df, title = ""){
  df$Z_STAT <- sign(df$BETA) * sqrt(df$CHISQ)
  ggplot(df,  aes(x = Z_STAT) ) +
  geom_histogram(aes(y = after_stat(density)), colour="black", fill="white", bins = 100) +
  stat_function(
    fun = dnorm, 
    col = "red",
    args = list(
      mean = mean(df$Z_STAT, na.rm = TRUE), 
      sd = sd(df$Z_STAT, na.rm = TRUE)
      )
  ) +
    theme_bw(16) +
    labs(title = title)
}
```

Now make histogram plot for each trait.
```{r, eval = FALSE}
# for Y1
plot.sumstats.hist(sumstats.y1, title = "Y1")
```

What do you observe as the case-control imbalance gets more severe?

4. Re-do 3 but now separate the histogram for common and rare SNPs. We define a new function to generate the histogram for each class of variants.
```{r, eval = FALSE}
plot.sumstats.hist.by.group <- function(df, title = ""){
  df$Z_STAT <- sign(df$BETA) * sqrt(df$CHISQ)
  df$group <- ifelse(grepl("rare", df$ID), "Rare SNPs", "Common SNPs")
  # Step 2: Generate normal density data for each group
  moment.ests <- with(df, tapply(Z_STAT, group, function(x) c(mean=mean(x, na.rm = TRUE), sd=sd(x, na.rm = TRUE))))
  z_stat_seq <- seq(min(df$Z_STAT, na.rm = TRUE), max(df$Z_STAT, na.rm = TRUE), length.out = 100)
  normal_curve_data <- do.call(rbind, lapply(unique(df$group), function(grp) {
    mean <- moment.ests[[grp]]['mean']; sd <- moment.ests[[grp]]['sd']
    density <- dnorm(z_stat_seq, mean = mean, sd = sd)
    data.frame(Z_STAT = z_stat_seq, density = density, group = grp)
  }))

  ggplot(df,  aes(x = Z_STAT) ) +
  geom_histogram(aes(y = ..density..), colour="black", fill="white", bins = 100) +
  geom_line(data = normal_curve_data, aes(x = Z_STAT, y = density), col = "red", size = 1) +
  facet_wrap(~group) +
  theme_bw(16) +
  labs(title = title)
}
```

* Make a histogram of the test statistics distribution at common/rare SNPs for each trait. What do you observe across the different case-control imbalances?
```{r, eval = FALSE}
# for Y1
plot.sumstats.hist.by.group(sumstats.y1, "Y1")
```

### *Extra*

5. Re-run GWAS in Questions 1 but now applying Firth correction. Make a QQ plot of the -log10 p-values for all 3 traits and eexamine at the histograms of the test statistic as in questions (3-4). 
```{r, eval = FALSE}
cmd <- sprintf('%s --bed cc_imb_geno --phenoFile cc_imb_pheno.txt --bt --step 2 --bsize 2000 --firth --approx --ignore-pred --out test_regenie_wFirth', regenie_binary)
system(cmd, intern = T)
```

This will produce three files (one for each phenotype) which you can read in R: 
```{r, eval = FALSE}
sumstats.y1.firth <- fread("test_regenie_wFirth_Y1.regenie") 
```

**<span style="color: red;"> To run the R implementation instead</span>**, run the following (this computes the association tests and stores it in the R variable directly)
```{r, eval = FALSE}
sumstats.y1.firth <- run_regenie_step2_bt(
  bedfile = "cc_imb_geno",
  phenofile = "cc_imb_pheno.txt",
  phenocol = "Y1",
  bsize = 300,
  firth = TRUE
) 
```

