---
title: "Practical 2 Key - Population Structure Inference"
author: ""
date: ""
output: html_document
editor_options:
  chunk_output_type: console
---
Before you begin:

* Make sure that R is installed on your computer
* For this lab, we will use the following R libraries:
```{r load-libs, message=FALSE, warning=FALSE}
library(data.table)
library(dplyr)
library(bigsnpr)
library(ggplot2)
```


## Introduction
We will be working with a subset of the genotype data from the Human Genome Diversity Panel (HGDP) and HapMap.

The file “[YRI_CEU_ASW_MEX_NAM.bed](https://github.com/joellembatchou/SISG2024_Association_Mapping/tree/master/data)” is a binary file in PLINK BED format with accompanying BIM and FAM files.
It contains the genotype data at autosomal SNPs (i.e. chromosomes 1-22) for:

* Native American samples from HGDP
* Four population samples from HapMap: 
  + Yoruba in Ibadan, Nigeria (YRI)
  + Utah residents with ancestry from
Northern and Western Europe (CEU)
  + Mexican Americans in Los Angeles, California (MXL)
  + African Americans from the south-western United States (ASW)

**File with ancestry labels assignment for each sample**: [Population_Sample_Info.txt](https://raw.githubusercontent.com/joellembatchou/SISG2024_Association_Mapping/master/data/Population_Sample_Info.txt) 

### Data preparation
Let’s first load the HGDP data into the R session. We need to define the path to the directory containing the PLINK BED and the ancestry label files (change the path to the file location).
```{r echo=FALSE}
# change this to the directory on your machine
HGDP_dir <- "/Users/joelle.mbatchou/SISG/2024/SISG2024_Association_Mapping/data/" 
plink2_binary <- "/Users/joelle.mbatchou/software/bins/plink2" 
```

```{r eval=FALSE}
# change this to the directory on your machine
HGDP_dir <- "/SISGM19/data/" 
```

Also specify the path to the PLINK2 binary
```{r eval=FALSE}
plink2_binary <- "/SISGM19/bin/plink2" 
```

We can now read the PLINK BED and FAM files (recall the BED file is a binary file):
```{r}
HGDP_bim <- fread(sprintf("%s/YRI_CEU_ASW_MEX_NAM.bim", HGDP_dir), header = FALSE)
head(HGDP_bim, 3)
HGDP_fam <- fread(sprintf("%s/YRI_CEU_ASW_MEX_NAM.fam", HGDP_dir), header = FALSE)
head(HGDP_fam, 3)
```

When reading the ancestry label file, we need to make sure the order of samples matches that in the PLINK data:
```{r}
HGDP_ancestry_df <- fread(sprintf("%s/Population_Sample_Info.txt", HGDP_dir))
HGDP_ancestry_df <- left_join(HGDP_fam[,c("V1","V2")], HGDP_ancestry_df, by = c("V1" = "FID", "V2" = "IID"))
head(HGDP_ancestry_df, 3)
```

## Exercises
Here are some things to look at:

1. Examine the dataset:
* How many samples are present? 

```{r}
str(HGDP_fam)
```

* How many SNPs?
```{r}
str(HGDP_bim)
```

* What is the number of samples in each population? 
```{r}
table(HGDP_ancestry_df$Population)
```

2. Get the first 10 principal components (PCs) in PLINK using all SNPs. 
```{r}
cmd <- sprintf("%s --bfile %s/YRI_CEU_ASW_MEX_NAM --pca 10 --out pca_plink", plink2_binary, HGDP_dir)
system(cmd)
```

This generates two files `pca_plink.eigenvec` containing the PCs (eigenvectors), and `pca_plink.eigenval` containing the top eigenvalues.

3. Read in the PCs in R and make a scatterplot of the first two PCs with each point colored by population membership.

```{r}
PC_df <- left_join(HGDP_ancestry_df, fread("pca_plink.eigenvec"), by = c("V1" = "#FID", "V2" = "IID"))
ggplot(PC_df, aes(x=PC1, y=PC2, color = Population)) +
  geom_point()
```

4. Interpret the first two PCs, what ancestries are they reflecting?

5. Read in the eigenvalues and make a scree plot corresponding for these first 10 PCs. Estimate the proportion of variance explained by the first two PCs.
```{r eigen-vals-comp}
eigenvalues_df <- fread("pca_plink.eigenval", header = FALSE)
ggplot(eigenvalues_df, aes(x = 1:10, y = V1)) +
  geom_point() +
  geom_line() +
  scale_x_continuous(breaks = 1:10) +
  labs(x = "PC", y = "Eigenvalue")
sum(eigenvalues_df$V1[1:2]) / sum(eigenvalues_df$V1)
```

6. Now redo Question 2 above using the [`bigsnpr` R package](https://privefl.github.io/bigsnpr/reference/index.html) specifying a $r^2$ threshold of 0.2 (i.e. LD pruning) as well as a minimum minor allele count (MAC) of 20.
```{r}
obj.bed <- bed(bedfile = sprintf("%s/YRI_CEU_ASW_MEX_NAM.bed", HGDP_dir))
pca.bigsnpr <- bed_autoSVD(
  obj.bed, 
  thr.r2 = 0.2, # R^2 threshold
  k = 10, # number of PCs
  min.mac = 20 # minimum minor allele count (MAC) filter
)
```
* You can evaluate the PCA results using
```{r bigsnpr-evals}
# plot PC2 vs PC1
plot(pca.bigsnpr, type = "scores", scores = 1:2)
# scree plot
plot(pca.bigsnpr) 
# plot SNP loadings for the first two PCs
plot(pca.bigsnpr, type = "loadings", loadings = 1:2, coeff = 0.4)
```

* Make a scatter plot of the first two principal components (PCs) with each point colored according to population membership. Does the plot change from the one in Question 2?
```{r plot-pca-bigsnpr}
plot(pca.bigsnpr, type = "scores", scores = 1:2) +
  aes(color = HGDP_ancestry_df$Population) +
  labs(color = "Population")
```

* Check the SNP loadings for the first 5 PCs.
```{r pc-five-loadings}
plot(pca.bigsnpr, type = "loadings", loadings = 1:5, coeff = 0.4)
``` 

7. Predict the proportion of Native American ancestry for the HapMap
MXL based on the PCA output in Question 3 *using one of the principal components* and assuming that the HapMap MXL have negligible African Ancestry. Which PC is most appropriate for this analysis?

*Hint*: To compute the average PC2 value for individuals of CEU ancestry
```{r}
ceu.mean <- with(PC_df, mean(PC2[Population == "CEU"]))
```

Do the same for individuals of NAM ancestry. How can you express distances of the MXL individuals relative to those means based on the chosen PC?

```{r}
nam.mean <- with(PC_df, mean(PC2[Population == "NAM"]))
mxl.prop.nam <- with(PC_df, (ceu.mean - PC2[Population == "MXL"]) / abs(ceu.mean - nam.mean))
summary(mxl.prop.nam)
```

8. Make a barplot of the proportional Native American/European ancestry estimates based on your output of question 7.

```{r}
sorted.props.nam <- sort(mxl.prop.nam, decreasing = TRUE)
df_mxl_props <- data.frame(
  anc.props = c(sorted.props.nam, 1 - sorted.props.nam),
  x = rep(1:length(sorted.props.nam), times = 2),
  population.labels = rep(c("NAM", "CEU"), each = length(sorted.props.nam))
)
ggplot(df_mxl_props, aes(x = x, y = anc.props, fill = population.labels)) +
  geom_bar(position="stack", stat="identity") +
  labs(x="Sample", y = "Ancestry Proportion", fill = "Population")
```

### *Extra*
9. Check if there are samples related 2nd degree or closer. If so, run PCA as in Question 6 removing these samples then project the remaining samples onto the PC space. Plot the top 2 PCs. 

* The command to check relatedness is:
```{r cache=TRUE, dependson = c(0)}
# check for 2nd degree relateds or closer
relatedness_info <- snp_plinkKINGQC(
  plink2.path = plink2_binary, 
  bedfile.in = sprintf("%s/YRI_CEU_ASW_MEX_NAM.bed", HGDP_dir), 
  thr.king = 2^-3.5, # threshold to identify 2nd degree relateds
  make.bed = FALSE
)
```
This returns a data frame which contains all pairs of individuals related 2nd degree or closer. 
```{r}
str(relatedness_info)
```

* We can then remove them when calling `bed_autoSVD()` using the `ind.row` argument.
```{r}
ind.rel <- match(c(relatedness_info$IID1, relatedness_info$IID2), obj.bed$fam$sample.ID) # relateds
ind.norel <- rows_along(obj.bed)[-ind.rel] # unrelateds
# Run PCA on unrelateds
obj.svd2 <- bed_autoSVD(
  obj.bed, 
  thr.r2 = 0.2, # R^2 threshold
  k = 10, # number of PCs
  min.mac = 20, # minimum minor allele count (MAC) filter
  ind.row = ind.norel
)
```

* Use `bed_projectSelfPCA()` to project related samples on the PC space.
*(Hint: This [tutorial document](https://privefl.github.io/bigsnpr/articles/bedpca.html) from `bigsnpr` will be helpful -- see the last section 'Project remaining individuals')*

```{r}
PCs <- matrix(NA, nrow(obj.bed), ncol(obj.svd2$u))
# PCs for unrelateds
PCs[ind.norel, ] <- predict(obj.svd2)
# Project relateds on PC space
proj <- bed_projectSelfPCA(
  obj.svd2, 
  obj.bed,
  ind.row = ind.rel
  )
PCs[ind.rel, ] <- proj$OADP_proj

# Plot the top 2 PCs with projections
pop_palette <- c("#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2")
names(pop_palette) <- unique(HGDP_ancestry_df$Population)

plot( # for unrelateds
  PCs[ind.norel, 1:2], 
  col = pop_palette[HGDP_ancestry_df$Population[ind.norel]], 
  pch = 1, xlab = "PC1", ylab = "PC2"
  )
points( # for relateds
  PCs[ind.rel, 1:2], 
  col = pop_palette[HGDP_ancestry_df$Population[ind.rel]], 
  pch = 2
  )
# add the legends
legend("topleft", legend = names(pop_palette), col = pop_palette, pch = 19, title = "Population")
legend("topright", legend = c("Model", "Projected"), col = c("black", "black"), pch = c(1, 2))
```

```{r echo=FALSE, message=FALSE, warning=FALSE, results='hide'}
# cleanup
file.remove(list.files(pattern = "pca_plink_*"))
```