---
title: "Practical 3 - GWAS in Samples with Structure & Using REGENIE"
author: ""
date: ""
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---
Before you begin:

* Make sure that R is installed on your computer
* For this lab, we will use the following R libraries:
```{r load-libs, message=FALSE, warning=FALSE}
library(data.table)
library(dplyr)
library(qqman)
library(ggplot2)
```



## Introduction
We will be analyzing a simulated data set which contains sample structure to better understand the impact it can have in GWAS analyses if not accounted for. 
We will perform GWAS on a quantitative phenotype which was simulated with high heritability and polygenic.

The file ["sim_rels_pheno.txt"](https://github.com/joellembatchou/SISG2025_Association_Mapping/tree/master/data)” contains the phenotype measurements for a set of individuals and the file "sim_rels_geno.bed" is a binary file in PLINK BED format with accompanying BIM and FAM files which contains the genotype data **at null variants** (i.e. not associated with the phenotype).  

*How should we expect the QQ/Manhatthan plots to look like under this scenario?*

### Data preparation
Let’s first load the simulated data into the R session. We need to define the path to the directory containing the phenotype and genotype files (change the path to the files location).
```{r echo=FALSE}
files_dir <- "/Users/joelle.mbatchou/SISG/2025/SISG2025_Association_Mapping/data/" 
plink2_binary <- "/Users/joelle.mbatchou/software/bins/plink2" 
regenie_binary <- "/Users/joelle.mbatchou/software/bins/regenie"
```

```{r eval=FALSE}
files_dir <- "/SISGM19/data/" 
```

Also specify the paths to the PLINK2 and REGENIE binaries:
```{r eval=FALSE}
plink2_binary <- "/SISGM19/bin/plink2" 
regenie_binary <- "/SISGM19/bin/regenie" 
```

We can now read the files (recall the PLINK BED file is a binary file):
```{r}
pheno_file <- fread(sprintf("%s/sim_rels_pheno.txt", files_dir), header = TRUE) 
head(pheno_file, 3)
sim_bim <- fread(sprintf("%s/sim_rels_geno.bim", files_dir), header = FALSE)
head(sim_bim, 3)
sim_fam <- fread(sprintf("%s/sim_rels_geno.fam", files_dir), header = FALSE)
head(sim_fam, 3)
```



## Exercises
Here are some things to try:

1. Examine the dataset:
  * How many samples are present? Use `str`
  * How many SNPs? In how many chromosomes? Use `str` and `table`

2. Examine the phenotype data:
  * How many individuals in the study have measurements? Use `table(is.na(pheno_file$Pheno))`
  * Plot a histogram to show the distribution of the phenotype. Use the `hist()` function

3. With PLINK, perform association mapping between the phenotype and the variants in the PLINK BED genotype file. Only perform association test on SNPs that pass the following quality control threshold filters:

* minor allele frequency (MAF) > 0.01
* at least a 99% genotyping call rate (less than 1% missing)
* HWE p-values greater than 0.001

The basic command would look like
```{r, eval = FALSE}
# first fill in the thresholds to use for each filter
filter_maf = 
filter_missing_rate = 
filter_hwe = 

cmd <- sprintf('%s --bfile "%s/sim_rels_geno" --pheno "%s/sim_rels_pheno.txt" --pheno-name Pheno --maf %g --geno %g --hwe %g --glm allow-no-covars --out gwas_plink', plink2_binary, files_dir, files_dir, filter_maf, filter_missing_rate, filter_hwe)
system(cmd, intern = T)
```
The results of the GWAS are stored in `gwas_plink.Pheno.glm.linear`.

4. Make a Manhattan plot of the association results. Make sure to check what information is stored in the PLINK output file (using `str()`).
```{r, eval = FALSE}
plink.gwas <- fread("gwas_plink.Pheno.glm.linear", header = TRUE)
plot(
  x = 1:nrow(plink.gwas),
  y = -log10(plink.gwas$P),
  col = c("orange", "purple")[1 + plink.gwas$`#CHROM` %% 2],
  xaxt = "n", xlab = "Genomic position", ylab = "Observed -log10(P)"
)
```

5. Make a Q-Q plot of the association results.
```{r, eval = FALSE}
qq(plink.gwas$P)
```

6. Compute the genomic control inflation factor $\lambda_{GC}$ based on the p-values. Is there evidence of possible inflation due to confounding?
```{r, eval = FALSE}
chisq.values <- qchisq(plink.gwas$P, 1, lower.tail = FALSE)
median(chisq.values)
```

7. Now we will run REGENIE to perform a GWAS of the phenotype using a whole genome regression model. We first want to extract a set of high quality variants for the Step 1 null model fitting. Using PLINK, apply QC filters to remove variants with MAF below 5%, missingness above 1%, HWE p-value below 0.001, minor allele count (MAC) below 20. 
We will use `--write-snplist` to store list of variants passing QC without making a new BED file.
```{r, eval = FALSE}
# first fill in the thresholds to use for each filter
filter_maf = 
filter_missing_rate = 
filter_hwe = 
filter_mac = 

cmd <- sprintf('%s --bfile "%s/sim_rels_geno" --pheno "%s/sim_rels_pheno.txt" --pheno-name Pheno --maf %g --geno %g --hwe %g --mac %g --write-snplist --out qc_pass', plink2_binary, files_dir, files_dir, filter_maf, filter_missing_rate, filter_hwe, filter_mac)
system(cmd, intern = T)
```
This produces a file `qc_pass.snplist` containing a list of variant IDs that pass the QC filters.
  
### If REGENIE software is installed on your machine
8. Run REGENIE Step 1 to fit the null model and obtain polygenic predictions using a leave-one-chromosome-out (LOCO) scheme. 
```{r, eval = FALSE}
cmd <- sprintf('%s --bed "%s/sim_rels_geno" --phenoFile "%s/sim_rels_pheno.txt"  --phenoCol Pheno --qt --step 1 --loocv --bsize 1000 --extract qc_pass.snplist --out gwas_regenie', regenie_binary, files_dir, files_dir)
system(cmd, intern = T)
```
The LOCO polygenic predictions for the phenotype are stored in `gwas_regenie_1.loco`.

9. Run REGENIE Step 2 to perform association testing.
```{r, eval = FALSE}
cmd <- sprintf('%s --bed "%s/sim_rels_geno" --phenoFile "%s/sim_rels_pheno.txt" --phenoCol Pheno --qt --step 2 --bsize 400 --pred gwas_regenie_pred.list --out step2_gwas_regenie', regenie_binary, files_dir, files_dir)
system(cmd, intern = T)
```  
The REGENIE summary statistics will be in `step2_gwas_regenie_Pheno.regenie`.

10. Generate Manhatthan and Q-Q plots based on the REGENIE association results and compute $\lambda_{GC}$. Compare with output from Questions 4-6.
  
### If REGENIE software does not run on your machine
8. We will use an implementation of REGENIE in R. Download it [here](https://gatech.box.com/s/k7yk3tpqlekuc7zh6xquyjh9vbvumcx7) and change the path of the variable `regenie_script` to the path of the script on your machine
```{r, eval = FALSE}
regenie_script <- "/Users/xyz/Downloads/run_regenie.r"
source(regenie_script)
```

* We now run REGENIE Step 1 to fit the null model and obtain polygenic predictions using a leave-one-chromosome-out (LOCO) scheme.
```{r, eval = FALSE}
loco_pred <- run_regenie_step1(
  bedfile = paste0(files_dir, "/sim_rels_geno"),
  phenofile = paste0(files_dir, "/sim_rels_pheno.txt"),
  phenocol = "Pheno",
  bsize = 1000,
  extract = "qc_pass.snplist"
) 
```
This function will return the LOCO polygenic predictions for the phenotype.

9. Run REGENIE Step 2 to perform association testing.
```{r, eval = FALSE}
sumstats_regenie <- run_regenie_step2(
  bedfile = paste0(files_dir, "/sim_rels_geno"),
  phenofile = paste0(files_dir, "/sim_rels_pheno.txt"),
  phenocol = "Pheno",
  bsize = 200,
  loco.mat = loco_pred
) 

str(sumstats_regenie)
```
This function returns a data frame containing the REGENIE summary statistics.

10. Generate Manhatthan and Q-Q plots based on the REGENIE association results and compute $\lambda_{GC}$. Compare with output from Questions 4-6.
  
